### ISSUE TITLE
setting p param in demo_pipeline

### ISSUE THREAD

**Issue by agiovann** [flatironinstitute/CaImAn#470]
For better support, please use the template below to submit your issue. When your issue gets resolved please remember to close it.

Sometimes errors while running CNMF occur during parallel processing which prevents the log to provide a meaningful error message.
Please reproduce your error with setting `dview=None`.

If you need to upgrade CaImAn follow the instructions given in the [wiki](https://github.com/flatironinstitute/CaImAn/wiki/Updating-CaImAn).

- **Tell us about your operating system (Linux/macOS/Windows), Python version (3.x), and working environment (Spyder/Jupyter Notebook/other)**

Mac OS X HIgh Sierra

- **Describe the issue that you are experiencing**

Something is changing the value of p in refit no matter what the input p is

- **Copy error log below**

In demo pipeline, using the lines
``` 
cnm.params.set('temporal', {'p': 0})
cnm2 = cnm.refit(images, dview=dview)
```
will then result in 
```
In[22]: cnm.params.temporal['p']
Out[22]: 2
```
and 
```
cnm2.params.temporal['p']
Out[24]: 2
```

- **If you're not reporting an error, type your message below**

**Comment by epnev** [comment]
fixed with f9fa2e2bf632ec38d4a7e456abf0f6da0b142f1f

**Comment by agiovann** [comment]
try to improve the clarity by not setting p inside and outside of the fit method

**Comment by epnev** [comment]
Addressed with 4c611263aa5972e6db0ca42ef721f3fc03ea5cc7 that sets a new parameter `params.patch['p_patch']` that controls the order of the AR dynamics within patch processing.
